src/Symfony/Bundle/FrameworkBundle/Routing/RouteCollectionBuilder.php

```diff
@@ -0,0 +1,465 @@
+     *
+     * @return RouteCollectionBuilder
+     */
+    public function import($resource, $prefix = null, $type = null)
+    {
+        /** @var RouteCollection $subCollection */
+        $subCollection = $this->resolve($resource, $type)->load($resource, $type);
+
+        return $this->addRouteCollection($subCollection, $prefix);
+    }
+
+    /**
+     * Adds a route and returns it for future modification.
+     *
+     * @param string      $path       The route path
+     * @param string      $controller The route controller string
+     * @param string|null $name       The name to give this route
+     *
+     * @return Route
+     */
+    public function add($path, $controller, $name = null)
+    {
+        $route = new Route($path);
+        $route->setDefault('_controller', $controller);
+        $this->addRoute($route, $name);
+
+        return $route;
+    }
+
+    /**
+     * Returns a RouteCollectionBuilder that can be configured and then added with addBuilder().
+     *
+     * @param string $prefix A prefix to apply to all routes added to this collection
+     *
+     * @return RouteCollectionBuilder
+     */
+    public function createBuilder($prefix = null)
+    {
+        $builder = new self($this->loader);
+        $builder->setPrefix($prefix);
+
+        return $builder;
+    }
+
+    /**
+     * Add a RouteCollectionBuilder.
+     *
+     * @param RouteCollectionBuilder $builder
+     */
+    public function addBuilder(RouteCollectionBuilder $builder)
+    {
+        $this->routes[] = $builder;
+    }
+
+    /**
+     * Adds a RouteCollection directly and returns those routes in a RouteCollectionBuilder.
+     *
+     * @param RouteCollection $collection
+     * @param string|null     $prefix
+     *
+     * @return $this
+     */
+    public function addRouteCollection(RouteCollection $collection, $prefix = null)
+    {
+        // create a builder from the RouteCollection
+        $builder = $this->createBuilder($prefix);
+        foreach ($collection->all() as $name => $route) {
+            $builder->addRoute($route, $name);
+        }
+
+        foreach ($collection->getResources() as $resource) {
+            $builder->addResource($resource);
+        }
+
+        $this->addBuilder($builder);
+
+        return $builder;
+    }
+
+    /**
+     * Adds a Route object to the builder.
+     *
+     * @param Route       $route
+     * @param string|null $name
+     *
+     * @return $this
+     */
+    public function addRoute(Route $route, $name = null)
+    {
+        if (null === $name) {
+            // used as a flag to know which routes will need a name later
+            $name = '_unnamed_route_'.spl_object_hash($route);
+        }
+
+        $this->routes[$name] = $route;
+
+        return $this;
+    }
+
+    /**
+     * Sets a prefix (e.g. /admin) to be used with all embedded routes.
+     *
+     * @param string $prefix
+     *
+     * @return $this
+     */
+    public function setPrefix($prefix)
```

Comment:
I don't think this one is useful at all. If you want a prefix, that's probably a subset of your routes that you've created via `createBuilder` anyway.